src/UI/Component/Trigger/Triggerer.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace ILIAS\UI\Component\Trigger;
+
+/**
+ * Interface Triggerer
+ *
+ * Any component that can trigger other components must implement this interface
+ */
+interface Triggerer
+{
+
+    /**
+     * Trigger an action of another component
+     *
+     * @param TriggerAction $action
+     * @param string $event
+     * @return self
+     */
+    public function triggerAction(TriggerAction $action, $event = 'click');
+
+
+    /**
+     * Get all actions of components triggered by this component
+     *
+     * @return TriggerAction[]
+     */
+    public function getTriggerActions();
```

Comment:
When rendering the triggerer aka button, I needed to get the triggered actions. In the current implementation, the triggerer is also responsible for rendering the triggered components (modals). We should discuss wether this behavior is desired or not.